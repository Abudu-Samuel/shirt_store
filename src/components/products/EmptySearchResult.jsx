import React from "react";

export default ({ query }) => (
  <div className="text-center">
    <h5 className="grey-text mb-4 mt-4">0 products found</h5>
    <h6 className="mb-4 grey-text">
      Unfortunately there was no match found for{" "}
      <span className="field-error">{query}</span>, please try again
    </h6>
    <h6 className="grey-text">
      Sorry, the page you have requested was not found. Try searching for a
      brand or keyword.
    </h6>
  </div>
);
