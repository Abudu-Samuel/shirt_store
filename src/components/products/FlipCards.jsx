import React from "react";
import ReactCardFlip from "react-card-flip";

export default ({
  isFlipped,
  handleCardFlip,
  handleProductCategories,
  handleProductDepartments
}) => (
  <div className="container">
    <div className="row">
      <div className="col-md-6">
        <ReactCardFlip isFlipped={isFlipped} flipDirection="horizontal">
          <div className="card front-card mb-2" key="front">
            <div className="card-body text-center">
              <h4 className="text-size">Categories.</h4>
            </div>
          </div>
          <div className="card mb-2" key="back">
            <div className="row mb-2 mt-2">
              <div className="col-md-12">
                {" "}
                <button
                  onClick={() => handleProductCategories(1)}
                  className="btn btn-sm btn-outline-default btn-dep"
                >
                  French
                </button>
                <button
                  onClick={() => handleProductCategories(2)}
                  className="btn btn-sm btn-outline-secondary btn-dep"
                >
                  Italian
                </button>
                <button
                  onClick={() => handleProductCategories(4)}
                  className="btn btn-sm btn-outline-info btn-dep"
                >
                  Animal
                </button>
                <button
                  onClick={() => handleProductCategories(3)}
                  className="btn btn-sm btn-outline-warning btn-dep"
                >
                  Irish
                </button>
                <button
                  onClick={() => handleProductCategories(5)}
                  className="btn btn-sm btn-outline-danger btn-dep"
                >
                  Flower
                </button>
                <button
                  onClick={() => handleProductCategories(6)}
                  className="btn btn-sm btn-outline-success btn-dep"
                >
                  Christmas
                </button>
                <button
                  onClick={() => handleProductCategories(7)}
                  className="btn btn-sm btn-outline-primary btn-dep"
                >
                  Valentine's
                </button>
              </div>
            </div>
          </div>
        </ReactCardFlip>
      </div>

      <div className="col-md-6">
        <ReactCardFlip isFlipped={isFlipped} flipDirection="horizontal">
          <div className="card front-card" key="front">
            <div className="card-body text-center">
              <h4 className="text-size">Departments.</h4>
            </div>
          </div>

          <div className="card" key="back">
            <div className="row mb-2 mt-2">
              <div className="col-md-12">
                {" "}
                <button
                  className="btn btn-sm btn-outline-default btn-dep"
                  disabled
                >
                  Autum
                </button>
                <button
                  onClick={() => handleProductDepartments(2)}
                  className="btn btn-sm btn-outline-secondary btn-dep"
                >
                  Nature
                </button>
                <button
                  className="btn btn-sm btn-outline-warning btn-dep"
                  disabled
                >
                  Rainy
                </button>
                <button
                  onClick={() => handleProductDepartments(3)}
                  className="btn btn-sm btn-outline-danger btn-dep"
                >
                  Seasonal
                </button>
                <button
                  className="btn btn-sm btn-outline-info btn-dep"
                  disabled
                >
                  Winter
                </button>
                <button
                  className="btn btn-sm btn-outline-success btn-dep"
                  disabled
                >
                  Sectional
                </button>
                <button
                  onClick={() => handleProductDepartments(1)}
                  className="btn btn-sm btn-outline-primary btn-dep"
                >
                  Regional
                </button>
              </div>
            </div>
          </div>
        </ReactCardFlip>
      </div>
    </div>

    <div className="col-10 col-md-4 mx-auto">
      <button
        className="btn btn-sm btn-outline-primary btn-dep"
        onClick={handleCardFlip}
      >
        view categories and departments
      </button>
    </div>
  </div>
);
