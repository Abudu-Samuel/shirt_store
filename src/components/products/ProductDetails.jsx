import React, { Component } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { Link } from "react-router-dom";
import StarRatings from "react-star-ratings";
import Loader from "../common/Loader";
import QuantityButton from "../common/QuantityButton";
import * as actionCreators from "../../redux/actions/actionCreators";

export class ProductDetails extends Component {
  constructor(props) {
    super(props);
    this.state = {
      rating: 2,
      quantity: 1,
      thumbnail: "",
      cart_id: "",
      product_id: "",
      attributes: ""
    };
  }

  componentDidUpdate(prevProps) {
    if (prevProps.product.product_id !== this.props.product.product_id) {
      this.setState({
        product_id: this.props.product.product_id,
        thumbnail: this.props.product.thumbnail
      });
    }
  }

  async componentDidMount() {
    this.props.actions.fetchSingleProduct(
      Number(this.props.match.params.product_id)
    );

    if (!localStorage.getItem("cartId")) {
      await this.props.actions.getCartId();
      this.setState({ cart_id: this.props.cartId.cart_id });
      localStorage.setItem("cartId", this.props.cartId.cart_id);
    } else {
      const cartID = localStorage.getItem("cartId");
      this.setState({ cart_id: cartID });
    }
  }

  changeRating(newRating, name) {
    this.setState({
      rating: newRating
    });
  }

  handleChange = event => {
    this.setState({
      [event.target.value]: event.target.name
    });
  };

  handleSize = size => {
    if (size === "xs") {
      this.setState({
        attributes: "xs"
      });
    }
    if (size === "s") {
      this.setState({
        attributes: "s"
      });
    }
    if (size === "m") {
      this.setState({
        attributes: "m"
      });
    }
    if (size === "l") {
      this.setState({
        attributes: "l"
      });
    }
    if (size === "xl") {
      this.setState({
        attributes: "xl"
      });
    }
    if (size === "xxl") {
      this.setState({
        attributes: "xxl"
      });
    }
  };

  changeImage = img => {
    this.setState({ thumbnail: img });
  };

  handleQuantity = arth => {
    if (arth === "+") {
      this.setState({
        quantity: this.state.quantity + 1
      });
    }

    if (arth === "-") {
      this.setState({
        quantity: this.state.quantity === 1 ? 1 : this.state.quantity - 1
      });
    }
  };

  addProductToCart = () => {
    const { cart_id, product_id, attributes } = this.state;
    const data = { cart_id, product_id, attributes };
    this.props.actions.addToCartAction(data, this.state.quantity);
  };

  render() {
    const { name, price, image, image_2, description } = this.props.product;

    return (
      <div className="container body">
        <div className="">
          {Object.keys(this.props.product).length < 1 ? (
            <Loader />
          ) : (
            <div className="row">
              <div className="col-md-1" />
              <div className="col-md-4 mt-5 mb-5">
                {this.state.thumbnail && (
                  <img
                    className="card-img-top img-fluid z-depth-3 mb-4"
                    src={require(`../../assets/images/product_images/${
                      this.state.thumbnail
                    }`)}
                    alt=""
                  />
                )}
                <div style={{ display: "flex", justifyContent: "center" }}>
                  <img
                    className="img-dt img-fluid z-depth-3 mr-4"
                    src={require(`../../assets/images/product_images/${image}`)}
                    onClick={() => this.changeImage(image)}
                    alt=""
                  />
                  <img
                    className="img-dt img-fluid z-depth-3"
                    src={require(`../../assets/images/product_images/${image_2}`)}
                    onClick={() => this.changeImage(image_2)}
                    alt=""
                  />
                </div>
              </div>
              <div className="col-md-6 offset-md-1 mt-5 mb-5">
                <p className="grey-text">
                  Home . All Categories . Men's clothings and accessories
                </p>
                <div>
                  <StarRatings
                    rating={2.403}
                    starRatedColor="gold"
                    starDimension="20px"
                    starSpacing="3px"
                  />
                </div>
                <h3 className="mt-1">{name}</h3>
                <h6 className="mt-1">{description}</h6>
                <h4 style={{ color: "#f7436b" }}>${price}</h4>
                <h6 className="grey-text mt-3 mb-3">Color</h6>
                <div className="mb-3">
                  <button
                    style={{ background: "#6EB2FB" }}
                    className="btn-circle hov btn-sm mr-3"
                    alt=""
                  />
                  <button
                    style={{ background: "#00D3CA" }}
                    className="btn-circle hov btn-sm mr-3"
                    alt=""
                  />
                  <button
                    style={{ background: "#F62F5E" }}
                    className="btn-circle hov btn-sm mr-3"
                    alt=""
                  />
                  <button
                    style={{ background: "#FE5C07" }}
                    className="btn-circle hov btn-sm mr-3"
                    alt=""
                  />
                  <button
                    style={{ background: "#F8E71C" }}
                    className="btn-circle hov btn-sm mr-3"
                    alt=""
                  />
                  <button
                    style={{ background: "#7ED321" }}
                    className="btn-circle hov btn-sm mr-3"
                    alt=""
                  />
                  <button
                    style={{ background: "#9013FE" }}
                    className="btn-circle hov btn-sm mr-3"
                    alt=""
                  />
                </div>
                <h6 className="grey-text mt-3 mb-3">Size</h6>
                <div className="mb-3">
                  <button
                    style={{
                      background:
                        this.state.attributes === "xs" ? "#f7436b" : "#EFEFEF",
                      color:
                        this.state.attributes === "xs" ? "#fff" : "#000000",
                      height: 30,
                      width: 62
                    }}
                    className="btn-sm mr-3 mb-2"
                    alt=""
                    onClick={() => this.handleSize("xs")}
                  >
                    XS
                  </button>
                  <button
                    style={{
                      background:
                        this.state.attributes === "s" ? "#f7436b" : "#EFEFEF",
                      color: this.state.attributes === "s" ? "#fff" : "#000000",
                      height: 30,
                      width: 62
                    }}
                    className="btn-sm mr-3"
                    alt=""
                    onClick={() => this.handleSize("s")}
                  >
                    S
                  </button>
                  <button
                    style={{
                      background:
                        this.state.attributes === "m" ? "#f7436b" : "#EFEFEF",
                      color: this.state.attributes === "m" ? "#fff" : "#000000",
                      height: 30,
                      width: 62
                    }}
                    className="btn-sm mr-3"
                    alt=""
                    onClick={() => this.handleSize("m")}
                  >
                    M
                  </button>
                  <button
                    style={{
                      background:
                        this.state.attributes === "l" ? "#f7436b" : "#EFEFEF",
                      color: this.state.attributes === "l" ? "#fff" : "#000000",
                      height: 30,
                      width: 62
                    }}
                    className="btn-sm mr-3"
                    alt=""
                    onClick={() => this.handleSize("l")}
                  >
                    L
                  </button>
                  <button
                    style={{
                      background:
                        this.state.attributes === "xl" ? "#f7436b" : "#EFEFEF",
                      color:
                        this.state.attributes === "xl" ? "#fff" : "#000000",
                      height: 30,
                      width: 62
                    }}
                    className="btn-sm mr-3"
                    alt=""
                    onClick={() => this.handleSize("xl")}
                  >
                    XL
                  </button>
                  <button
                    style={{
                      background:
                        this.state.attributes === "xxl" ? "#f7436b" : "#EFEFEF",
                      color:
                        this.state.attributes === "xxl" ? "#fff" : "#000000",
                      height: 30,
                      width: 62
                    }}
                    className="btn-sm mr-3"
                    alt=""
                    onClick={() => this.handleSize("xxl")}
                  >
                    XXL
                  </button>
                </div>
                <h6 className="grey-text mt-3 mb-3">Quantity</h6>
                <QuantityButton
                  handleQuantity={this.handleQuantity}
                  handleChange={this.handleChange}
                  quantity={this.state.quantity}
                />
                <button
                  style={{
                    background: "#f7436b",
                    color: "#ffffff",
                    borderRadius: 50
                  }}
                  className="btn btn-large btn-round"
                  onClick={this.addProductToCart}
                  data-toggle="modal"
                  data-target="#modalAbandonedCart"
                >
                  Add to cart
                </button>

                {/* start */}
                <div
                  className="modal fade right"
                  id="modalAbandonedCart"
                  tabIndex="-1"
                  role="dialog"
                  aria-labelledby="myModalLabel"
                  aria-hidden="true"
                  data-backdrop="false"
                >
                  <div
                    className="modal-dialog modal-dialog-centered modal-notify modal-info"
                    role="document"
                  >
                    <div className="modal-content">
                      <div className="modal-header">
                        <p className="heading">Product in the cart</p>

                        <button
                          type="button"
                          className="close"
                          data-dismiss="modal"
                          aria-label="Close"
                        >
                          <span aria-hidden="true" className="white-text">
                            &times;
                          </span>
                        </button>
                      </div>

                      <div className="modal-body">
                        <div className="row">
                          <div className="col-md-3">
                            <p />
                            <p className="text-center">
                              <i className="fas fa-shopping-cart fa-4x" />
                            </p>
                          </div>

                          <div className="col-md-9">
                            <p>
                              Do you need more time to make a purchase decision?
                            </p>
                            <p>
                              No pressure, your product will be waiting for you
                              in the cart.
                            </p>
                          </div>
                        </div>
                      </div>

                      <div className="modal-footer justify-content-center">
                        <Link to="/cart">
                          <button className="btn btn-md btn-info">
                            Go to cart
                          </button>
                        </Link>
                        <Link to="/">
                          <button className="btn btn-md btn-outline-info waves-effect">
                            continue shopping
                          </button>
                        </Link>
                      </div>
                    </div>
                  </div>
                </div>
                {/* end */}
              </div>
              <div className="col-md-1" />
            </div>
          )}
        </div>
      </div>
    );
  }
}

const mapStateToProps = ({
  products: { product, cartId, cart, updateQuantity }
}) => ({
  product,
  cartId,
  cart,
  updateQuantity
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(actionCreators, dispatch)
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ProductDetails);
