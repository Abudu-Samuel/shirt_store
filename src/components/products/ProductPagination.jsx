import React from "react";
import ReactPaginate from "react-paginate";

export default ({ handlePagination, pages }) => (
  <div>
    <ReactPaginate
      previousLabel="Previous"
      nextLabel="Next"
      onPageChange={handlePagination}
      pageCount={Math.ceil(pages / 8)}
      containerClassName="pagination pagination-sm custom-pagination"
      pageLinkClassName="page-link"
      nextLinkClassName="page-link next"
      previousLinkClassName="page-link previous"
      disabledClassName="disabled"
      pageClassName="page-item"
      previousClassName="page-item"
      nextClassName="page-item"
      activeClassName="active"
      subContainerClassName="pages pagination"
    />
  </div>
);
