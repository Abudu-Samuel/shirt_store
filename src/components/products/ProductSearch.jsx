import React, { Component } from "react";
import { connect } from "react-redux";
import Loader from "../common/Loader";
import ReactPaginate from "react-paginate";
import { Link } from "react-router-dom";
import { bindActionCreators } from "redux";
import EmptySearchResult from "./EmptySearchResult";
import * as actionCreators from "../../redux/actions/actionCreators";

class ProductSearch extends Component {
  componentDidMount() {
    this.props.actions.searchProducts(this.props.match.params.query, 1);
  }

  componentWillUnmount() {
    this.props.actions.searchProducts("", 1);
  }

  handlePagination = page => {
    const nextPage = page.selected + 1;
    const { actions } = this.props;
    return actions.searchProducts(this.props.match.params.query, nextPage);
  };

  handleSingleProduct = id => {
    const productId = this.props.searchProducts.rows.filter(
      product => product.product_id === id
    );
    this.props.actions.fetchSingleProduct(productId[0].product_id);
  };

  render() {
    const { searchProducts } = this.props;
    return (
      <div className="container">
        <div className="row loader">
          {Object.keys(searchProducts).length < 1 || this.props.isFetching ? (
            <Loader />
          ) : searchProducts.rows.length > 0 || this.props.isFetching ? (
            searchProducts.rows.map(product => (
              <div key={product.product_id} className="col-md-3 mb-3 mt-3">
                <Link to={`/products/${product.product_id}`}>
                  <div
                    className="card product-card"
                    onClick={() => this.handleSingleProduct(product.product_id)}
                  >
                    <img
                      className="card-img-top img-fluid"
                      src={require(`../../assets/images/product_images/${
                        product.thumbnail
                      }`)}
                      alt=""
                    />

                    <div className="card-body">
                      <h6 className="text-center">
                        {product.name.length >= 20
                          ? product.name.slice(0, 17).concat("...")
                          : product.name}
                      </h6>
                      <span className="text-left" style={{ color: "#f7436b" }}>
                        <strike>${product.price}</strike>
                      </span>
                      <span style={{ float: "right", color: "#f7436b" }}>
                        ${product.discounted_price}
                      </span>
                    </div>
                  </div>
                </Link>
              </div>
            ))
          ) : (
            <EmptySearchResult query={this.props.match.params.query} />
          )}
        </div>
        {Object.keys(searchProducts).length < 1 ? null : (
          <ReactPaginate
            previousLabel="Previous"
            nextLabel="Next"
            onPageChange={this.handlePagination}
            pageCount={Math.ceil(this.props.pages / 4)}
            containerClassName="pagination pagination-sm custom-pagination"
            pageLinkClassName="page-link"
            nextLinkClassName="page-link next"
            previousLinkClassName="page-link previous"
            disabledClassName="disabled"
            pageClassName="page-item"
            previousClassName="page-item"
            nextClassName="page-item"
            activeClassName="active"
            subContainerClassName="pages pagination"
          />
        )}
      </div>
    );
  }
}

const mapStateToProps = ({
  products: { searchProducts, isFetching, pages }
}) => ({
  searchProducts,
  isFetching,
  pages
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(actionCreators, dispatch)
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ProductSearch);
