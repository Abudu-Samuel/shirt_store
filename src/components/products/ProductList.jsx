import React from "react";
import { Link } from "react-router-dom";
import Loader from "../common/Loader";
import ProductPagination from "./ProductPagination";

export default ({
  products,
  handlePagination,
  pages,
  isFetching,
  handleSingleProduct,
  handleProductCategories,
  catInDept
}) => {
  return (
    <div className="container">
      <div className="">
        {Object.keys(products).length < 1 || isFetching ? (
          <Loader />
        ) : (
          <div className="row">
            <div className="col-md-2 mt-3">
              <h4>Categories</h4>
              <hr />
              {Object.keys(catInDept).length < 1 ? (
                <div>
                  <p
                    onClick={() => handleProductCategories(1)}
                    className="mt-4 category"
                  >
                    French
                  </p>
                  <p
                    onClick={() => handleProductCategories(2)}
                    className="category"
                  >
                    Italian
                  </p>
                  <p
                    onClick={() => handleProductCategories(3)}
                    className="category"
                  >
                    Irish
                  </p>
                  <p
                    onClick={() => handleProductCategories(4)}
                    className="category"
                  >
                    Animal
                  </p>
                  <p
                    onClick={() => handleProductCategories(5)}
                    className="category"
                  >
                    Flower
                  </p>
                  <p
                    onClick={() => handleProductCategories(6)}
                    className="category"
                  >
                    Christmas
                  </p>
                  <p
                    onClick={() => handleProductCategories(7)}
                    className="category"
                  >
                    Valentine's
                  </p>
                </div>
              ) : (
                <div>
                  {catInDept.map(category => (
                    <div>
                      <p
                        onClick={() =>
                          handleProductCategories(category.category_id)
                        }
                        className="mt-4 category"
                      >
                        {category.name}
                      </p>
                    </div>
                  ))}
                </div>
              )}
            </div>
            <div className="col-md-10">
              <div className="row">
                {products.rows.map(product => (
                  <div key={product.product_id} className="col-md-4 mb-3 mt-3">
                    <Link to={`/products/${product.product_id}`}>
                      <div
                        className="card product-card"
                        onClick={() => handleSingleProduct(product.product_id)}
                      >
                        <img
                          className="card-img-top img-fluid"
                          src={require(`../../assets/images/product_images/${
                            product.thumbnail
                          }`)}
                          alt=""
                        />

                        <div className="card-body">
                          <h6 className="text-center">
                            {product.name.length >= 20
                              ? product.name.slice(0, 17).concat("...")
                              : product.name}
                          </h6>
                          {product.discounted_price === "0.00" ? (
                            <span
                              className="d-flex justify-content-center"
                              style={{ color: "#f7436b" }}
                            >
                              {product.price}
                            </span>
                          ) : (
                            <div>
                              <span
                                className="text-left"
                                style={{ color: "#f7436b" }}
                              >
                                <strike>${product.price}</strike>
                              </span>
                              <span
                                style={{ float: "right", color: "#f7436b" }}
                              >
                                ${product.discounted_price}
                              </span>
                            </div>
                          )}
                        </div>
                      </div>
                    </Link>
                  </div>
                ))}
              </div>
            </div>
          </div>
        )}
      </div>
      <div className="row">
        <div className="col-lg-12">
          {Object.keys(products).length < 1 ? null : (
            <ProductPagination
              handlePagination={handlePagination}
              pages={pages}
            />
          )}
        </div>
      </div>
    </div>
  );
};
