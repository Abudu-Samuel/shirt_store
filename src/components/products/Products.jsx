import React, { Component } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import * as actionCreators from "../../redux/actions/actionCreators";
import Carousel from "../common/Carousel";
import sliderOne from "../../assets/images/icons/bag/slide-ex.jpg";
import sliderTwo from "../../assets/images/icons/bag/slide-ez.jpg";
import sliderThree from "../../assets/images/icons/bag/men.jpg";
import ProductList from "./ProductList";

export class Products extends Component {
  state = {
    id: ""
  };

  componentDidMount() {
    this.props.actions.fetchAllProducts(1);
  }

  handlePagination = page => {
    const nextPage = page.selected + 1;
    const { category, actions, department } = this.props;
    if (category) {
      return actions.fetchProductCategory(category, nextPage);
    } else if (department) {
      return actions.fetchProductDepartment(department, nextPage);
    }
    return actions.fetchAllProducts(nextPage);
  };

  handleCardFlip = event => {
    event.preventDefault();
    this.setState({ isFlipped: !this.state.isFlipped });
  };

  handleProductCategories = id => {
    this.props.actions.fetchProductCategory(id, 1);
  };

  handleProductDepartments = id => {
    this.props.actions.fetchProductDepartment(id, 1);
  };

  handleSingleProduct = id => {
    const productId = this.props.products.rows.filter(
      product => product.product_id === id
    );
    this.props.actions.fetchSingleProduct(productId[0].product_id);
  };

  render() {
    const { products, pages, isFetching, catInDept } = this.props;
    const slides = [
      <img className="img-fluid slide-img" src={sliderTwo} alt="1" />,
      <img className="img-fluid slide-img" src={sliderOne} alt="2" />,
      <img className="img-fluid slide-img" src={sliderThree} alt="3" />
    ];
    return (
      <div>
        <Carousel className="container" slides={slides} />

        <h4 className="text-center">Departments</h4>
        <hr />
        <p
          style={{ cursor: "pointer", color: "#f7436b" }}
          className="text-center"
        >
          <span
            onClick={() => this.handleProductDepartments(1)}
            className="mr-4"
          >
            Regional
          </span>{" "}
          <span
            onClick={() => this.handleProductDepartments(2)}
            className="mr-4"
          >
            Nature
          </span>
          <span onClick={() => this.handleProductDepartments(3)}>Seasonal</span>
        </p>
        <ProductList
          products={products}
          handlePagination={this.handlePagination}
          pages={pages}
          isFetching={isFetching}
          handleSingleProduct={this.handleSingleProduct}
          handleProductCategories={this.handleProductCategories}
          catInDept={catInDept}
          id={this.state.id}
        />
      </div>
    );
  }
}

const mapStateToProps = ({
  products: { products, category, pages, department, isFetching, catInDept }
}) => ({
  products,
  category,
  pages,
  department,
  isFetching,
  catInDept
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(actionCreators, dispatch)
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Products);
