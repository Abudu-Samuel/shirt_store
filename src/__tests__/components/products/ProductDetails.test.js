import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import { ProductDetails } from "../../../components/products/ProductDetails";

const props = {
  product: {
    name: "Arc d'Triomphe",
    price: "14.99",
    image: "arc-d-triomphe-thumbnail.gif",
    image_2: "arc-d-triomphe-2.gif",
    description: "This beautiful and iconic T-shirt"
  },
  cartId: 101,
  match: {
    params: 1
  },
  actions: {
    fetchSingleProduct: jest.fn(() => Promise.resolve()),
    getCartId: jest.fn(() => Promise.resolve())
  }
};

const event = {
  target: {
    quantity: 2
  }
};

describe("<ProductDetails />", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<ProductDetails {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("invokes handleChange method", () => {
    const wrapper = shallow(<ProductDetails {...props} />);
    const handleChangeSpy = jest.spyOn(wrapper.instance(), "handleChange");
    wrapper.instance().handleChange(event);
    expect(handleChangeSpy).toHaveBeenCalled();
  });
});
