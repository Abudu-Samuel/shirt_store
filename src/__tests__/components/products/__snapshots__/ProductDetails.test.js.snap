// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDetails /> renders without crashing 1`] = `
<div
  className="container body"
>
  <div
    className=""
  >
    <div
      className="row"
    >
      <div
        className="col-md-1"
      />
      <div
        className="col-md-4 mt-5 mb-5"
      >
        <div
          style={
            Object {
              "display": "flex",
              "justifyContent": "center",
            }
          }
        >
          <img
            alt=""
            className="img-dt img-fluid z-depth-3 mr-4"
            onClick={[Function]}
            src="arc-d-triomphe-thumbnail.gif"
          />
          <img
            alt=""
            className="img-dt img-fluid z-depth-3"
            onClick={[Function]}
            src="arc-d-triomphe-2.gif"
          />
        </div>
      </div>
      <div
        className="col-md-6 offset-md-1 mt-5 mb-5"
      >
        <p
          className="grey-text"
        >
          Home . All Categories . Men's clothings and accessories
        </p>
        <div>
          <StarRatings
            changeRating={null}
            gradientPathName=""
            ignoreInlineStyles={false}
            numberOfStars={5}
            rating={2.403}
            starDimension="20px"
            starEmptyColor="rgb(203, 211, 227)"
            starHoverColor="rgb(230, 67, 47)"
            starRatedColor="gold"
            starSpacing="3px"
            svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
            svgIconViewBox="0 0 51 48"
            typeOfWidget="Star"
          />
        </div>
        <h3
          className="mt-1"
        >
          Arc d'Triomphe
        </h3>
        <h6
          className="mt-1"
        >
          This beautiful and iconic T-shirt
        </h6>
        <h4
          style={
            Object {
              "color": "#f7436b",
            }
          }
        >
          $
          14.99
        </h4>
        <h6
          className="grey-text mt-3 mb-3"
        >
          Color
        </h6>
        <div
          className="mb-3"
        >
          <button
            alt=""
            className="btn-circle hov btn-sm mr-3"
            style={
              Object {
                "background": "#6EB2FB",
              }
            }
          />
          <button
            alt=""
            className="btn-circle hov btn-sm mr-3"
            style={
              Object {
                "background": "#00D3CA",
              }
            }
          />
          <button
            alt=""
            className="btn-circle hov btn-sm mr-3"
            style={
              Object {
                "background": "#F62F5E",
              }
            }
          />
          <button
            alt=""
            className="btn-circle hov btn-sm mr-3"
            style={
              Object {
                "background": "#FE5C07",
              }
            }
          />
          <button
            alt=""
            className="btn-circle hov btn-sm mr-3"
            style={
              Object {
                "background": "#F8E71C",
              }
            }
          />
          <button
            alt=""
            className="btn-circle hov btn-sm mr-3"
            style={
              Object {
                "background": "#7ED321",
              }
            }
          />
          <button
            alt=""
            className="btn-circle hov btn-sm mr-3"
            style={
              Object {
                "background": "#9013FE",
              }
            }
          />
        </div>
        <h6
          className="grey-text mt-3 mb-3"
        >
          Size
        </h6>
        <div
          className="mb-3"
        >
          <button
            alt=""
            className="btn-sm mr-3 mb-2"
            onClick={[Function]}
            style={
              Object {
                "background": "#EFEFEF",
                "color": "#000000",
                "height": 30,
                "width": 62,
              }
            }
          >
            XS
          </button>
          <button
            alt=""
            className="btn-sm mr-3"
            onClick={[Function]}
            style={
              Object {
                "background": "#EFEFEF",
                "color": "#000000",
                "height": 30,
                "width": 62,
              }
            }
          >
            S
          </button>
          <button
            alt=""
            className="btn-sm mr-3"
            onClick={[Function]}
            style={
              Object {
                "background": "#EFEFEF",
                "color": "#000000",
                "height": 30,
                "width": 62,
              }
            }
          >
            M
          </button>
          <button
            alt=""
            className="btn-sm mr-3"
            onClick={[Function]}
            style={
              Object {
                "background": "#EFEFEF",
                "color": "#000000",
                "height": 30,
                "width": 62,
              }
            }
          >
            L
          </button>
          <button
            alt=""
            className="btn-sm mr-3"
            onClick={[Function]}
            style={
              Object {
                "background": "#EFEFEF",
                "color": "#000000",
                "height": 30,
                "width": 62,
              }
            }
          >
            XL
          </button>
          <button
            alt=""
            className="btn-sm mr-3"
            onClick={[Function]}
            style={
              Object {
                "background": "#EFEFEF",
                "color": "#000000",
                "height": 30,
                "width": 62,
              }
            }
          >
            XXL
          </button>
        </div>
        <h6
          className="grey-text mt-3 mb-3"
        >
          Quantity
        </h6>
        <_default
          handleChange={[Function]}
          handleQuantity={[Function]}
          quantity={1}
        />
        <button
          className="btn btn-large btn-round"
          data-target="#modalAbandonedCart"
          data-toggle="modal"
          onClick={[Function]}
          style={
            Object {
              "background": "#f7436b",
              "borderRadius": 50,
              "color": "#ffffff",
            }
          }
        >
          Add to cart
        </button>
        <div
          aria-hidden="true"
          aria-labelledby="myModalLabel"
          className="modal fade right"
          data-backdrop="false"
          id="modalAbandonedCart"
          role="dialog"
          tabIndex="-1"
        >
          <div
            className="modal-dialog modal-dialog-centered modal-notify modal-info"
            role="document"
          >
            <div
              className="modal-content"
            >
              <div
                className="modal-header"
              >
                <p
                  className="heading"
                >
                  Product in the cart
                </p>
                <button
                  aria-label="Close"
                  className="close"
                  data-dismiss="modal"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                    className="white-text"
                  >
                    Ã—
                  </span>
                </button>
              </div>
              <div
                className="modal-body"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-md-3"
                  >
                    <p />
                    <p
                      className="text-center"
                    >
                      <i
                        className="fas fa-shopping-cart fa-4x"
                      />
                    </p>
                  </div>
                  <div
                    className="col-md-9"
                  >
                    <p>
                      Do you need more time to make a purchase decision?
                    </p>
                    <p>
                      No pressure, your product will be waiting for you in the cart.
                    </p>
                  </div>
                </div>
              </div>
              <div
                className="modal-footer justify-content-center"
              >
                <Link
                  to="/cart"
                >
                  <button
                    className="btn btn-md btn-info"
                  >
                    Go to cart
                  </button>
                </Link>
                <Link
                  to="/"
                >
                  <button
                    className="btn btn-md btn-outline-info waves-effect"
                  >
                    continue shopping
                  </button>
                </Link>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-1"
      />
    </div>
  </div>
</div>
`;
